examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,430 @@
+mae = median_absolute_error(y_test, y_pred)
+r2score = model.score(X_test, y_test)
+
+string_score += '\nMAE on testing set: {0:.2f} $/hour'.format(mae)
+string_score += '\nR2 score: {0:.4f}'.format(r2score)
+fig, ax = plt.subplots(figsize=(6, 6))
+sns.regplot(y_test, y_pred)
+
+plt.text(3, 20, string_score)
+
+plt.ylabel('Model predictions')
+plt.xlabel('Truths')
+plt.xlim([0, 27])
+plt.ylim([0, 27])
+
+##############################################################################
+# The model learnt is far from being a good model making accurate predictions:
+# the R squared score is very low.
+# As interpretation tools characterize model rather than the generative process
+# of the data itself, it needs to be emphasized that interpretations are
+# correct if the model is correct as well.
+# In this case, we are more interested in providing a methodology than in
+# having a good description of the data: a bad example could illustrate the
+# importance of cross checking the results.
+#
+# Interpreting coefficients
+# .........................
+#
+# First of all, we can plot the values of the coefficients of the regressor we
+# have fitted.
+
+feature_names = (model.named_steps['columntransformer']
+                      .named_transformers_['onehotencoder']
+                      .get_feature_names(input_features=categorical_columns))
+feature_names = np.concatenate(
+    [feature_names, binary_columns, numerical_columns])
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# Soon we realize that we cannot compare different coefficients since the
+# features have different natural scales and hence value ranges
+# because of their different unit of measure.
+# For instance, the AGE coefficient is expressed in $/hours/living years
+# while the EDUCATION is expressed in $/hours/years of education.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std(axis=0).plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# We should then multiply the coefficients by the standard deviation and we
+# will be able to better compare them: in that way, we emphasize that the
+# greater the variance of a feature, the larger the impact of the corresponding
+# coefficent on the output, all else being equal.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_ *
+    X_train_preprocessed.std(axis=0),
+    columns=['Coefficients'], index=feature_names
```

Comment:
The legend of the plot should be e.g. 'Coefficients * std dev' instead of 'Coefficients'

Reply:
What about "Coefficient importance" or "Feature Coefficient importance"?

Reply:
Then, could we imagine to add to linear regressors the attribute `feature_importance_` defined as above? @GaelVaroquaux?